### PR TITLE
[ATIP] Initial uiautomator as backend in ATIP

### PR DESCRIPTION
**Pull request by Honry**
- Add uiautomator libraries and steps
 - Initial uiautomator class
 BUG=https://crosswalk-project.org/jira/browse/XWALK-4652

### CODE REVIEW

**Comment by Honry** [comment]
@yugang , @zqzhang , please help to review this PR.

File: tools/atip/atip/android/android.py [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+# ARE DISCLAIMED. IN NO EVENT SHALL INTEL CORPORATION BE LIABLE FOR ANY DIRECT,
+# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# Authors:
+#         Yang, Yunlong <yunlongx.yang@intel.com>
```
Comment (zqzhang) [inline tools/atip/atip/android/android.py]: ?? The original author?

File: tools/atip/atip/android/steps.py [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+# ARE DISCLAIMED. IN NO EVENT SHALL INTEL CORPORATION BE LIABLE FOR ANY DIRECT,
+# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# Authors:
+#         Yang, Yunlong <yunlongx.yang@intel.com>
```
Comment (zqzhang) [inline tools/atip/atip/android/steps.py]: ditto

File: tools/atip/atip/web/web.py [hunk 3]
```diff
@@ -748,11 +749,14 @@ def launch_webapp_by_name(
     if not context.bdd_config:
         assert False
 
-    if app_name in context.webs:
-        context.webs[app_name].quit()
-    context.webs.update(
+    if app_name in context.apps:
```
Comment (zqzhang) [inline tools/atip/atip/web/web.py]: Hmm, this has been changed by #2526, please resolve the conflict.

File: tools/atip/tests/environment.py [hunk 4]
```diff
@@ -24,30 +24,31 @@
 # EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #
 # Authors:
-#         Fan, Yugang <yugang.fan@intel.com>
+#         Yang, Yunlong <yunlongx.yang@intel.com>
```
Comment (zqzhang) [inline tools/atip/tests/environment.py]: Then who is the author, by the way?

**Comment by zqzhang** [comment]
Sorry, I am still not clear about the ATIP working flow, could you please give me a brief introduction offline?

File: tools/atip/atip/android/android.py (hunk 1 above)
Comment (Honry) [inline tools/atip/atip/android/android.py]: Yes, Yunlong help draft most coding work.

File: tools/atip/atip/web/web.py (hunk 3 above)
Comment (Honry) [inline tools/atip/atip/web/web.py]: OK, I will.

File: tools/atip/tests/environment.py (hunk 4 above)
Comment (Honry) [inline tools/atip/tests/environment.py]: Yugang is the original author, I will update it.

**Comment by Honry** [comment]
Of cause, let's talk offline.

**Comment by Honry** [comment]
Solve the conflict.

**Comment by Honry** [comment]
UT result: Run all BDD tests, get the same result with old ATIP.

**Pull request merged**